src/os/shared.rs

```diff
@@ -19,6 +19,21 @@ use crate::os::lsof::get_open_sockets;
 #[cfg(target_os = "windows")]
 use crate::os::windows::get_open_sockets;
 
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq)]
+pub struct ProcessInfo {
+    pub name: String,
+    pub pid: i32,
```

Comment:
yeah, that is what i thought but then [fn pid()](https://docs.rs/procfs/latest/procfs/process/struct.Process.html#impl-Process-2) of `procfs` returns an i32 so i thought it would be better to conform to that. but anyway, after a quick search i found that PID's can't be negative so i changed the struct definition to u32 now.